src/app/components/action-button.ts

```diff
@@ -0,0 +1,18 @@
+/// <reference path="../../../typings/tsd.d.ts" />
+
+import {Component, View, ElementRef} from 'angular2/angular2';
+
+@Component({
+	selector: 'action-button'
+})
+@View({
+	template: `<button (click)="doAction()">Action!</button>`
+})
+export class ActionButton {
+	constructor(public el:ElementRef) {
+		
+	}
+	doAction(){
+		this.el.domElement.dispatchEvent(new Event('yummy'));
```

Comment:
rather than using native events you should use angular's version.

```es6
@Component({
  selector: 'action-button'
  events: ['click']
})
@View({
  template: `<button (click)="doAction($event)">Action!</button>`
})
export class ActionButton {
	constructor(public el:ElementRef) {
		
	}
	doAction(event) {
		this.click.next({ data: 'yolo', event: event })
  }
}
```